multus/multus.go

```diff
@@ -417,23 +426,30 @@ func cmdDel(args *skel.CmdArgs, exec invoke.Exec, kubeClient k8s.KubeClient) err
 
 	numK8sDelegates, kc, err := k8s.TryLoadPodDelegates(k8sArgs, in, kubeClient)
 	if err != nil {
-		return err
+		// in error case, need to proceed without return for bailout
+		_ = logging.Errorf("Multus: Err in getting k8s network from pod: %v", err)
 	}
 
 	if numK8sDelegates == 0 {
+		logging.Debugf("cmdDel: reading delegates from cache")
 		// re-read the scratch multus config if we have only Multus delegates
```

Comment:
Hmm. Shouldn't we also cover the case of the pod annotation changing before DEL? Unless we admission-control it, the user can delete a network from the selection annotation, and then multus won't clean that network up on DEL.

Perhaps we should do something like:

1) cache all network configs (even cluster-wide default network) on ADD
2) on DEL, tear down the cluster-wide default network. Use the on-disk config for this (because it too could have changed), but if for some reason that doesn't exist, fall back to /etc/cni/net.d or Net Attach object config.
3) Do a readdir() in the sandbox container ID's cache directory to find all other networks to delete from the cache info, and then delete them.
4) only if we cannot find the cached info for the sandbox container ID, ask Kubernetes what the annotations are and use that to determine what networks to delete.

Basically, after a pod ADD we cannot trust the kube API not to change.

Reply:
Agree, fix it and update PR diff.

Reply:
@dcbw Considering the case, I come up with the idea to 'just use cache in deletion' because not only pod annotation, also defaultNetwork/clusterNetwork may be changed during the pod running. In case of dynamic change of network interface, multus re-cache to track the changes then we can make cache consistent. I will change it and update PRs.

Reply:
But we still need to keep to use k8sclients in cmdDel() to update networkstatus...